src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/CompoundTargetFrameworkValueProvider.cs

```diff
@@ -70,21 +70,30 @@ public override Task<bool> IsValueDefinedInContextAsync(string propertyName, IPr
                 // Changing the Target Framework Moniker
                 if (StringComparers.PropertyLiteralValues.Equals(propertyName, InterceptedTargetFrameworkProperty))
                 {
-                    // Delete stored platform properties
-                    storedProperties.TargetPlatformIdentifier = null;
-                    storedProperties.TargetPlatformVersion = null;
-                    await ResetPlatformPropertiesAsync(defaultProperties);
-
+                    if (Strings.IsNullOrEmpty(unevaluatedPropertyValue))
+                    {
+                        return null;
+                    }
+                    
                     storedProperties.TargetFrameworkMoniker = unevaluatedPropertyValue;
+                    
+                    // Only projects targeting .NET 5 or higher use platform properties.
+                    string targetFrameworkAlias = await GetTargetFrameworkAliasAsync(unevaluatedPropertyValue);
+                    if (!IsNetCore5OrHigher(targetFrameworkAlias))
+                    {
+                        // Delete platform properties
+                        storedProperties.TargetPlatformIdentifier = null;
+                        storedProperties.TargetPlatformVersion = null;
+                        await ResetPlatformPropertiesAsync(defaultProperties);
+                    }
                 }
 
                 // Changing the Target Platform Identifier
                 else if (StringComparers.PropertyLiteralValues.Equals(propertyName, TargetPlatformProperty))
                 {
                     if (unevaluatedPropertyValue != storedProperties.TargetPlatformIdentifier)
                     {
-                        // Delete stored platform properties
-                        storedProperties.TargetPlatformIdentifier = null;
+                        // Delete platform properties.
                         storedProperties.TargetPlatformVersion = null;
```

Comment:
Above we set both `TargetPlatformIdentifier` and `TargetPlatformVersion` to null, but here it's only the latter. Is that intentional?

Reply:
That's right: this is the case when the user is setting a platform identifier different from what they had, i.e. windows -> android, so we reset whatever we had as the target platform version, in the windows example we probably had a 10.0.